### ISSUE TITLE
Rename sendShortcut and implement new API

### ISSUE THREAD

**Issue by alexdavid** [Originate/nifty#22]
From our talk today with @kevgo and @dhakiki

* Rename `sendShortcut` to `pressKeys`
* New API:
```coffeescript
.pressKeys 'a b'      # should press the keys a and b together
.pressKeys '[cmd]s'   # should press keys command and s together
.pressKeys '⌥[space]' # should press keys alt and space together
.pressKeys '[CTRL]+v' # should press keys control and v together
```

Some notes I'm thinking about:
* Characters in square brackets should be case-insensitive
* `+` and whitespace should be ignored
* special characters should be escaped with backslash: `\+`, `\[`, `\]`, `\\`

What do you think?

**Comment by kevgo** [comment]
I think "-" is also commonly used to denote that keys are pressed together: `[alt]-n`.

Also, how do we handle uppercase characters outside of square brackets? Are they treated as character + [SHIFT]? Example: is `.pressKeys 'A'` the same as `.pressKeys 'a'`, or as `.pressKeys '[SHIFT]-a'`? My vote goes to the latter.

**Comment by alexdavid** [comment]
Yeah I think uppercase letters should be a shortcut for shift + that letter

**Comment by dhakiki** [comment]
Small issue with the backslash. since the first back slash is ignored, the user would be required to enter two.

<img width="276" alt="screen shot 2015-07-28 at 12 21 43 pm" src="https://cloud.githubusercontent.com/assets/2328865/8941252/5be7e218-3523-11e5-8b12-581af3feda5b.png">

Are we ok with that @alexdavid @kevgo ?

**Comment by kevgo** [comment]
What would that be? Key "a", "\", and "b" pressed together. I think that's such an extreme edge case that I'm fine with the current behavior.

**Comment by alexdavid** [comment]
It is a problem I didn't think about. And we should come up with a better solution because obviously isn't clear because it confused Kevin.

`"a\\+b"` would be the keys a, +, and b together. But this get's really bad if we want to do something like `ctrl+\` which would be `"[ctrl]+\\\\"` which is ridiculous. We could introduce a `[backslash]` keyword to make that slightly better.

The main problem is that we are proposing using the same escape character as javascript uses for strings. We could just use our bracket notation to escape our special characters
for example:
`[ctrl] + [plus]` is `CTRL` and `+`
`[ctrl] + [open-bracket]` is `CTRL` and `[`
`[ctrl] + [backslash]` is `CTRL` and `\`
`[ctrl] + [space]` is `CTRL` and ` `

**Comment by alexdavid** [comment]
What if instead of doing this string parsing we just take in a variable amount of arguments?

```coffeescript
.pressKeys 'a', 'b'       # should press the keys a and b together
.pressKeys '[cmd]', 's'   # should press keys command and s together
.pressKeys '⌥', '[space]' # should press keys alt and space together
.pressKeys '[CTRL]', 'v'  # should press keys control and v together
```

**Comment by dhakiki** [comment]
YESS -- I like that! ^^ 

An array of inputs would be way easier to work with since before I was trying to figure out a way to deal with all the edge cases mentioned above (not entering spaces, entering spaces, entering +, entering -). It wasn't clean.

**Comment by kevgo** [comment]
I'm game! An array is the most natural way of providing several keys. Lets start with that. If that is ever not enough, we can the later come up with some sort of DSL for describing key combinations through strings.
